### PR TITLE
Added Step-By-Step Guide For Simulator

### LINKED ISSUE #377: Step-by-step guide for simulator
It would be a good feature if we add step by step guide for simulator like a website tour.

**Additional context**
 
guide

**Are you working on this? (Yes/No)**
Yes

### PR DESCRIPTION
**Pull request by Sj1218305**
Fixes  #377 #622 #953 #955 

#### Describe the changes you have made in this PR -
Added a Step-By-Step Guide For Simulator. If the user will open the Simulator for the very first time it will automatically pop up . Also, he can view that guide whenever he wants to from the guide button in header.

### CODE REVIEW

File: app/views/simulator/edit.html.erb [hunk 1]
```diff
@@ -186,7 +195,7 @@
                   <span class="glyphicon glyphicon-arrow-down" aria-hidden="true"></span>
                 </button>
                 <button type="button" class="btn zoomButton btn-lg" onclick="layoutTitleLeft()">
-                  <span class="glyphicon glyphicon-arrow-left" aria-hidden="true"></span>
+                  <span class="glyphicon glyphicon-arrow-left" aria-hidden="thrrue"></span>
```
Comment (satu0king) [inline app/views/simulator/edit.html.erb]: What is `thrrue` ?

File: app/views/simulator/edit.html.erb [hunk 2]
```diff
@@ -1,13 +1,15 @@
...
 <!-- <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha384&#45;BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous"> -->
+<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/intro.js/2.9.3/introjs.css">
 <link rel="stylesheet" href="https://code.jquery.com/ui/1.12.1/themes/base/jquery-ui.css">
 <script>
-  userSignedIn = <%= user_signed_in? %>;
-  logix_project_id = <%= @logix_project_id %>;
-  projectName = "<%= @projectName %>"
+  userSignedIn = <%= user_signed_in?.to_json %>;
+  logix_project_id = <%= @logix_project_id.to_json %>;
+  projectName = "<%= @projectName.to_json %>"
```
Comment (satu0king) [inline app/views/simulator/edit.html.erb]: Why `to_json` ? is it not working without that? @tachyons is this safer?

File: app/views/logix/index.html.erb [hunk 3]
```diff
@@ -171,7 +172,19 @@
...
+<script>
+function clickCounter() {
+    if(typeof(Storage) !== "undefined") {
+      if (localStorage.clickcount) {
+        localStorage.clickcount = Number(localStorage.clickcount)+1;
+      } else {
+        localStorage.clickcount = 1;
 
+        }
+    }
```
[the remark this answers is not in the document]
Comment (satu0king) [reply app/views/logix/index.html.erb]: What is this click counter?

**Comment by satu0king** [comment]
@Sj1218305, @Shivansh2407 is right 
image 
When I loaded first, the initial guide was at the bottom of the page (The user cannot even move further). And yes, there are some color issues which we can fix later. However, we cannot show the guide every time to the user. An easy solution to this is use `localStorage` and store a flag whether the preview has already been shown or not. Probably a better way is to use cookies.

File: app/views/layouts/_header.html.erb [hunk 4]
```diff
@@ -56,3 +56,4 @@
     </ul>
   </div>
 </nav>
+<script src="/js/utils.js"></script>
```
Comment (tachyons) [inline app/views/layouts/_header.html.erb]: Why ?

**Comment by abstrekt** [comment]
since the UI is being revamped, this PR won't be valid anymore? @satu0king @nitin10s

**Pull request closed without merging**